<launch>


  <arg name='agent_names' value='antonio elias ivan'/>
  <param name='agent_names' value='$(arg agent_names)'/>

  <rosparam param='target_position'>
    [1.0, -1.0]
  </rosparam>

  <!--remap from='antonio/neighbor_bearing_measurement' to='elias/bearing_measurement'/-->
  <!--remap from='elias/neighbor_bearing_measurement' to='antonio/bearing_measurement'/-->

  <group ns='antonio'>

    <rosparam param='initial_position'>
      [0.0, 0.0]
    </rosparam>

    <rosparam param='target_position'>
      [1.0, -1.0]
    </rosparam>

    <rosparam command='load' file='$(find circumnavigation-planner)/param/simulator.yaml'/>

    <remap from='neighbor_bearing_measurement' to='/elias/bearing_measurement'/>

    <node name='vehicle_simulator'
      pkg='circumnavigation-planner'
      type='vehicle_simulator.py'/>

    <rosparam param='initial_estimate'>
      [1.0, 1.0]
    </rosparam>

    <node name='planner'
      pkg='circumnavigation-planner'
      type='planner.py'/>

    <node name='sensor_simulator'
      pkg='circumnavigation-planner'
      type='sensor_simulator.py'/>

  </group>


  <group ns='elias'>

    <rosparam param='initial_position'>
      [0.0, 2.0]
    </rosparam>

    <rosparam param='target_position'>
      [1.0, -1.0]
    </rosparam>

    <rosparam command='load' file='$(find circumnavigation-planner)/param/simulator.yaml'/>

    <remap from='neighbor_bearing_measurement' to='/antonio/bearing_measurement'/>

    <node name='vehicle_simulator'
      pkg='circumnavigation-planner'
      type='vehicle_simulator.py'/>

    <rosparam param='initial_estimate'>
      [1.0, 1.0]
    </rosparam>

    <node name='planner'
      pkg='circumnavigation-planner'
      type='planner.py'/>

    <node name='sensor_simulator'
      pkg='circumnavigation-planner'
      type='sensor_simulator.py'/>

  </group>


  <rosparam command='load' file='$(find circumnavigation-planner)/param/plotter.yaml'/>

  <node name='plotter'
    pkg='circumnavigation-planner'
    type='plotter.py'/>


</launch>
